### ISSUE TITLE
CORS Plugin Access-Control-Allow-Headers returned only if Access-Control-Request-Method is specified

### ISSUE THREAD

**Issue by hvindin** [Kong/kong#8978]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

2.8.1

### Current Behavior

With the CORS plugin configured:

Send a request at kong without an Access-Control-Request-Method header:
```bash
curl http://127.0.0.1:8880/healthcheck -X OPTIONS -H 'Origin: https://something.example.com/' -i -H 'Access-Control-Request-Headers: Some-Header,Some-Other-Header'
```

And the request just travels on through to the backend system and isn't handled as a preflight request (all the other returned headers except vary, ACAO, and ACAC, are from the backend):

```
HTTP/1.1 200 OK
Content-Type: application/json
Content-Length: 20
Connection: keep-alive
Server: nginx
Date: Mon, 20 Jun 2022 02:21:31 GMT
Cache-Control: no-cache
Expires: -1
X-UA-Compatible: IE=Edge
X-Frame-Options: DENY
Content-Security-Policy: frame-ancestors 'none'; upgrade-insecure-requests
X-XSS-Protection: 1;mode=block
Referrer-Policy: strict-origin
Strict-Transport-Security: max-age=63072000; includeSubdomains; preload
vary: Origin
Access-Control-Allow-Origin: https://something.example.com/
Access-Control-Allow-Credentials: true
```

### Expected Behavior

A CORS response should be returned.

The mozilla developer docs here:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Headers
State that:
> The preflight request is an OPTIONS request that includes some combination of the three preflight request headers: Access-Control-Request-Method, Access-Control-Request-Headers, and Origin.

However CORS headers only appear to be returned if the Access-Control-Request-Method header is specified.

### Steps To Reproduce

I am using declarative kong configuration, this is the kong.yml file I am using:

```
_format_version: "2.1"
services:
- name: healthcheck
  host: localhost
  port: 8080
  protocol: http
  connect_timeout: 30000
  read_timeout: 907500
  retries: 5
  write_timeout: 30000
  routes:
  - paths:
    - /healthcheck
    path_handling: v0
    preserve_host: true
    protocols:
    - http
    regex_priority: 0
    strip_path: false
    https_redirect_status_code: 426
  connect_timeout: 30000
  read_timeout: 907500
  retries: 5
  write_timeout: 30000

plugins:
- name: cors
  config:
    origins:
      - '^https://.*\.example\.com\.au$'
      - '^https://.*\.example\.com$'
    credentials: true
    max_age: 3600
    preflight_continue: false
```

Send a request at kong like so:
```
curl http://127.0.0.1:8880/healthcheck -X OPTIONS -H 'Origin: https://something.example.com' -i -H 'Access-Control-Request-Headers: Some-Header,Some-Other-Header' -H 'Access-Control-Request-Method: POST'
```
Everything looks fine:
```
HTTP/1.1 200 OK
Date: Mon, 20 Jun 2022 02:21:26 GMT
Connection: keep-alive
vary: Origin
Access-Control-Allow-Origin: https://something.example.com
Access-Control-Allow-Credentials: true
Access-Control-Allow-Headers: Some-Header,Some-Other-Header
Access-Control-Allow-Methods: GET,HEAD,PUT,PATCH,POST,DELETE,OPTIONS,TRACE,CONNECT
Access-Control-Max-Age: 3600
Content-Length: 0
```

Send a request at kong without an Access-Control-Request-Method header:
```
curl http://127.0.0.1:8880/healthcheck -X OPTIONS -H 'Origin: https://something.example.com' -i -H 'Access-Control-Request-Headers: Some-Header,Some-Other-Header'
```

And the request just travels on through to the backend system and isn't handled as a preflight request (all the other returned headers except vary, ACAO, and ACAC, are from the backend):
```
HTTP/1.1 200 OK
Content-Type: application/json
Content-Length: 20
Connection: keep-alive
Server: nginx
Date: Mon, 20 Jun 2022 02:21:31 GMT
Cache-Control: no-cache
Expires: -1
X-UA-Compatible: IE=Edge
X-Frame-Options: DENY
Content-Security-Policy: frame-ancestors 'none'; upgrade-insecure-requests
X-XSS-Protection: 1;mode=block
Referrer-Policy: strict-origin
Strict-Transport-Security: max-age=63072000; includeSubdomains; preload
vary: Origin
Access-Control-Allow-Origin: https://something.example.com
Access-Control-Allow-Credentials: true
```

**Comment by samugi** [comment]
This should be fine and expected, it was introduced [here](https://github.com/Kong/kong/commit/40095104cbf3732f25f679e020361b360af49af7) to fix [#4898](https://github.com/Kong/kong/issues/4898). The header [Access-Control-Request-Method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Request-Method) is required in an actual preflight request and is being used by the CORS plugin ([along with the OPTIONS method and the Origin header](https://github.com/Kong/kong/blob/59ed75961b71a23a592117965e508b6ac0c4bde4/kong/plugins/cors/handler.lua#L189-L194)) to distinguish requests that should be proxied to upstream from those that should be handled as preflight.

**Comment by hvindin** [comment]
Thanks samugi, that makes sense after reading #4898.
